src/main/java/io/featureflow/client/core/RestClientImpl.java

```diff
@@ -191,6 +191,10 @@ public long getRetryInterval() {
                 // Do not retry if over max retry count
                 return false;
             }
+            if (exception instanceof ConnectTimeoutException) {
```

Comment:
Because ConnectTimeoutException extends InterruptedIOException, we have to move it to before InterruptedIOException else it will never be caught